app-frontend/src/app/components/settings/addPhotoModal/addPhotoModal.module.js

```diff
@@ -0,0 +1,193 @@
+import angular from 'angular';
+import addPhotoModalTpl from './addPhotoModal.html';
+
+
+const AddPhotoModalComponent = {
+    templateUrl: addPhotoModalTpl,
+    controller: 'AddPhotoModalController',
+    bindings: {
+        close: '&',
+        dismiss: '&',
+        modalInstance: '<',
+        resolve: '<'
+    }
+};
+
+class AddPhotoModalController {
+    constructor($scope, Upload, organizationService) {
+        'ngInject';
+
+        this.$scope = $scope;
+
+        this.organizationService = organizationService;
+        this.Upload = Upload;
+    }
+
+    $onInit() {
+        this.initSteps();
+        this.organizationId = this.resolve.organizationId;
+    }
+
+    initSteps() {
+        this.steps = [];
+        this.steps = this.steps.concat([{
+            name: 'UPLOAD',
+            previous: () => false,
+            allowPrevious: () => false,
+            next: () => 'PREVIEW',
+            allowNext: () => this.selectedPhoto,
```

Comment:
I found it hard to tell if I successfully selected an image because there was no indication that my selection had worked. I think we could automatically advance to the next step after a valid image is selected to make the flow a bit more user-friendly.